reconcile/prometheus_rules_tester/integration.py

```diff
@@ -0,0 +1,266 @@
+import json
+import logging
+import sys
+from collections.abc import Mapping
+from dataclasses import dataclass
+from typing import (
+    Any,
+    Optional,
+)
+
+import yaml
+from sretoolbox.utils import threaded
+
+import reconcile.openshift_resources_base as orb
+from reconcile import queries
+from reconcile.status import ExitCodes
+from reconcile.utils import (
+    gql,
+    promtool,
+)
+from reconcile.utils.semver_helper import make_semver
+from reconcile.utils.structs import CommandExecutionResult
+
+# This comes from prometheus-operator. It is the largest configmap that they will
+# create. It is also the largest PrometheusRule yaml file they will process to add it
+# to the configmap.
+MAX_CONFIGMAP_SIZE = 0.5 * 1024 * 1024
+
+QONTRACT_INTEGRATION = "prometheus_rules_tester"
+QONTRACT_INTEGRATION_VERSION = make_semver(0, 1, 0)
+
+PROVIDERS = ["prometheus-rule"]
+
+NAMESPACE_NAME = "openshift-customer-monitoring"
+
+
+@dataclass
+class TestContent:
+    test_path: str
+    test: dict
+
+
+@dataclass
+class Test:
+    cluster: str
+    namespace_name: str
+    rule_path: str
+    rule: dict
+    rule_length: int
+    tests: Optional[list[TestContent]]
+    result: Optional[CommandExecutionResult] = None
+
+
+@dataclass
+class RuleToFetch:
+    namespace: dict[str, Any]
+    resource: dict[str, Any]
+
+
+PROMETHEUS_RULES_TESTS_QUERY = """
+{
+  tests: resources_v1(schema: "/app-interface/prometheus-rule-test-1.yml") {
+    path
+    content
+  }
+}
+"""
+
+
+# this is just for the early exit
+def get_prometheus_tests() -> dict[str, str]:
+    """Returns a path indexed dict with the prometheus tests content"""
+    gqlapi = gql.get_api()
+    tests = {}
+    for t in gqlapi.query(PROMETHEUS_RULES_TESTS_QUERY)["tests"]:
+        # This can be a jinja template. We cannot load the yaml here
+        tests[t["path"]] = t["content"]
+
+    return tests
```

Comment:
Since this is a new integration, can we use generated queries here? I assume you need a dict format for compat with `orb`. In that case, we could still use `.dict(by_alias=True)`. This will make refactoring of orb easier later on, as this integration will then already support typed queries.

Reply:
fixed in 2d04214b